api/src/test/java/org/openmrs/AllergiesTest.java

```diff
@@ -330,15 +332,14 @@ public void addAll_shouldNotAllowDuplicateNonCodedAllergen(){
 		
 		Allergies allergies = new Allergies();
 		allergies.add(allergy1);
-		allergies.add(allergy2);
-		
-		allergies.addAll(allergies);
+		assertThrows(APIException.class, () -> allergies.add(allergy2));
+		assertThrows(APIException.class, () -> allergies.addAll(allergies));
```

Comment:
@achilep you changed the meaning of this test.

When converting `@Test(expected` to assertThrows you should only have to delete 1 line. The line that throws the exception since you wrap it in assertThrows

```java
		Allergies allergies = new Allergies();
		allergies.add(allergy1);
		allergies.add(allergy2);
		
		assertThrows(APIException.class, () -> allergies.addAll(allergies));
```

Please check the PR, after having pushed and go through my comments. If you see a line where I comment please dont delete and its still highlighted it means you haven't resolved my comment.

Reply:
i have this problem with this problem:
![Screenshot from 2020-07-05 07-39-45](https://user-images.githubusercontent.com/39851246/86527052-044f5780-be93-11ea-938a-119ba5439850.png)

![Screenshot from 2020-07-05 07-39-29](https://user-images.githubusercontent.com/39851246/86527054-09140b80-be93-11ea-977c-678621c9ac45.png)

Reply:
sorry @teleivo  for the format problem, I solved all this problem locally, but I don't understand how online it is something else.

Reply:
on my computer
![Screenshot from 2020-07-05 09-07-05](https://user-images.githubusercontent.com/39851246/86528380-3e265b00-be9f-11ea-8004-1ced0c79b577.png)
online
![Screenshot from 2020-07-05 09-06-13](https://user-images.githubusercontent.com/39851246/86528355-1931e800-be9f-11ea-9423-5ad20c396784.png)

Reply:
That could be because of tabs/spaces. we use tabs in OpenMRS. You can enable the IDE to show whitespaces to see if its that issue. Then replace all spaces with tabs.

Reply:
> @achilep you changed the meaning of this test.
> 
> When converting `@Test(expected` to assertThrows you should only have to delete 1 line. The line that throws the exception since you wrap it in assertThrows
> 
> ```java
> 		Allergies allergies = new Allergies();
> 		allergies.add(allergy1);
> 		allergies.add(allergy2);
> 		
> 		assertThrows(APIException.class, () -> allergies.addAll(allergies));
> ```
> 
> Please check the PR, after having pushed and go through my comments. If you see a line where I comment please dont delete and its still highlighted it means you haven't resolved my comment.

So you are saying that

`allergies.add(allergy2);` already throws the APIException.class.

Then you are right to **only** assert on

`assertThrows(APIException.class, () -> allergies.add(allergy2));`

Sorry for misleading you then. Please remove the other assertThrows and only keep the above.

Reply:
> sorry @teleivo for the format problem, I solved all this problem locally, but I don't understand how online it is something else.

That could be because of tabs/spaces. we use tabs in OpenMRS. You can enable the IDE to show whitespaces to see if its that issue. Then replace all spaces with tabs.

Or maybe you fixed it locally but did not push your changes. Or you did not commit the changes that fixed the problem.